packages/core/src/components/tabs/index.js

```diff
@@ -0,0 +1,104 @@
+class Tabs {
+  static instances = new WeakMap();
+
+  static get cssClasses() {
+    return {
+      ACTIVE: 'ray-tabs__item--active',
+      ACTIVE_CONTENT: 'ray-tabs__content--active'
+    };
+  }
+
+  static get strings() {
+    return {
+      INIT_SELECTOR: '.ray-tabs'
+    };
+  }
+
+  static create(element, options) {
+    return this.instances.get(element) || new this(element, options);
+  }
+
+  static createAll(target = document, _options = {}) {
+    const options = {
+      initSelector: this.strings.INIT_SELECTOR,
+      ..._options
+    };
+
+    const tabs = Array.from(target.querySelectorAll(options.initSelector));
+    tabs.forEach(tabGroup => this.create(tabGroup, options));
+  }
+
+  constructor(root, options) {
+    this._root = root;
+    this._options = options;
+    this.constructor.instances.set(this._root, this);
+
+    this.tabItems = Array.from(this._root.querySelectorAll('.ray-tabs__item'));
+    this.contentItems = Array.from(
+      this._root.querySelectorAll('.ray-tabs__content')
```

Comment:
i would suggest using a data- identifier for the content and subsequently querying the entire dom for the content portion. This way if we want to use tabs but have the content appear in a ancestor element it will be possible